schema.ts

```diff
@@ -1203,6 +1211,44 @@ export interface ObjectSchemaDefinition<K extends string, L extends string>
   modifiedByProperty?: PropertyIdentifier<K>;
 
   // TODO(dweitzman): Only support options in the typing when the codaType is ValueHintType.SelectList.
+
+  // TODO(patrick): Unhide this.
+  /**
+   * This property should be typed as either an Email or a Person.
+   * @hidden
+   */
+  permissionUserProperty?: K;
+
+  // TODO(patrick): Unhide this.
+  /**
+   * This property should be typed one of: an Email, a Person, or a reference to a sync table which has
+   * isUserPrincipalTable=true.
+   * @hidden
+   */
+  permissionGroupMembersProperty?: K;
+}
+
+// TODO(patrick): Unhide this
+/** @hidden */
+export interface Permission {
+  // The simplest ACL is just a user email.
+  userEmail?: string;
+
+  // If there is an isUserPrincipal table, you can use the idProperty of its rows to reference them.
+  userReference?: string | number;
+
+  // If more than 1 sync table isGroupPrincipal, this specifies the identityName of the group
+  // being referenced.
+  groupIdentityName?: string;
```

Comment:
If a user can be in multiple groups, how are they specified?  Each group must have a `Permission`?  Should any of the groups properties be arrays?

Reply:
A user's membership in groups is defined by the isGroupPrincipalTable's permission property, not here.

Here's an example of what I'm imagining one item could return as its permission list:
```
[
  {userEmail: 'creator@foo.com'},
  {groupIdentityName: 'Team', groupReference: 'project-x-team'},
  {groupIdentityName: 'Team', groupReference: 'project-y-team'},
  {groupIdentityName: 'Department', groupReference: 'engineering'}
]
```

Not necessarily the most efficient, I grant you.